### PR TITLE
Check list

### PR DESCRIPTION
**Pull request by Daarii**

### CODE REVIEW

File: src/modules/checklists/components/AddButton.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    const { afterSave } = this.props;
+
+    if (afterSave) {
+      afterSave();
+    }
+
+    this.overlayTrigger.hide();
+  };
+
+  renderForm(props) {
```
Comment (munkhjin0223) [inline src/modules/checklists/components/AddButton.tsx]: delete unnecessary "props"

File: src/modules/checklists/components/AddButton.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  afterSave?: () => void;
+};
+
+class ChecklistAdd extends React.Component<IProps> {
+  private overlayTrigger;
+
+  hidePopover = () => {
+    const { afterSave } = this.props;
+
+    if (afterSave) {
```
Comment (munkhjin0223) [inline src/modules/checklists/components/AddButton.tsx]: ?

File: src/modules/checklists/components/AddButton.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import { __ } from 'modules/common/utils';
+import * as React from 'react';
+import { OverlayTrigger, Popover } from 'react-bootstrap';
+import AddChecklistForm from '../containers/AddChecklistForm';
+import { PopoverContent } from '../styles';
+
+type IProps = {
+  item: IItem;
+  options: IOptions;
+  afterSave?: () => void;
```
Comment (munkhjin0223) [inline src/modules/checklists/components/AddButton.tsx]: delete it

File: src/modules/checklists/components/AddButton.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import { IItem, IOptions } from 'modules/boards/types';
+import { __ } from 'modules/common/utils';
+import * as React from 'react';
+import { OverlayTrigger, Popover } from 'react-bootstrap';
+import AddChecklistForm from '../containers/AddChecklistForm';
+import { PopoverContent } from '../styles';
+
+type IProps = {
+  item: IItem;
+  options: IOptions;
```
Comment (munkhjin0223) [inline src/modules/checklists/components/AddButton.tsx]: replace with "type"

File: src/modules/checklists/components/AddChecklistForm.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+type State = {
+  disabled: boolean;
+  title: string;
+};
+
+class AddChecklistForm extends React.Component<IProps, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
```
Comment (munkhjin0223) [inline src/modules/checklists/components/AddChecklistForm.tsx]: add space

File: src/modules/checklists/components/AddChecklistForm.tsx [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+          <Button
+            disabled={this.state.disabled}
+            btnStyle="success"
+            icon="checked-1"
+            type="submit"
+            size="small"
+          >
+            Save
+          </Button>
```
Comment (munkhjin0223) [inline src/modules/checklists/components/AddChecklistForm.tsx]: replace with renderButton common function

File: src/modules/checklists/graphql/mutations.ts [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+`;
+
+// checklist items
+
+const commonItemVariables = `
+  $checklistId: String,
+  $isChecked: Boolean,
+  $content: String,
+  $mentionedUserIds: [String],
+  $order: Int
```
Comment (munkhjin0223) [inline src/modules/checklists/graphql/mutations.ts]: delete

File: src/modules/checklists/graphql/mutations.ts [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+`;
+
+const commonItemParams = `
+  checklistId: $checklistId,
+  isChecked: $isChecked,
+  content: $content,
+  mentionedUserIds: $mentionedUserIds,
+  order: $order
```
Comment (munkhjin0223) [inline src/modules/checklists/graphql/mutations.ts]: delete

File: src/modules/checklists/graphql/mutations.ts [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+  mutation checklistItemsAdd(
+    ${commonItemVariables}
+  ) {
+    checklistItemsAdd(
+      ${commonItemParams}
+    ) {
+      _id
+      isChecked
+      content
+      order
```
Comment (munkhjin0223) [inline src/modules/checklists/graphql/mutations.ts]: delete

File: src/modules/checklists/graphql/mutations.ts [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+    ${commonItemVariables}
+  ) {
+    checklistItemsEdit(
+      _id: $_id,
+      ${commonItemParams}
+    ) {
+      _id
+      isChecked
+      content
+      order
```
Comment (munkhjin0223) [inline src/modules/checklists/graphql/mutations.ts]: delete

File: src/modules/checklists/graphql/mutations.ts [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+  }
+`;
+
+const updateOrderItems = `
+  mutation updateOrderItems($orders: [OrderInput]) {
+    updateOrderItems(orders: $orders) {
+      _id
+    }
+  }
+`;
```
Comment (munkhjin0223) [inline src/modules/checklists/graphql/mutations.ts]: delete

File: src/modules/checklists/graphql/queries.ts [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+  contentType
+  contentTypeId
+  title
+  createdUserId
+  createdDate
+  items {
+    _id
+    isChecked
+    content
+    order
```
Comment (munkhjin0223) [inline src/modules/checklists/graphql/queries.ts]: delete

File: src/modules/checklists/graphql/queries.ts [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+      contentTypeId
+      title
+      createdUserId
+      createdDate
+      items {
+        _id
+        checklistId
+        isChecked
+        content
+        order
```
Comment (munkhjin0223) [inline src/modules/checklists/graphql/queries.ts]: delete

File: src/modules/checklists/styles.ts [hunk 14]
```diff
@@ -0,0 +1,128 @@
+import { colors, dimensions } from 'modules/common/styles';
+import styled from 'styled-components';
+import styledTS from 'styled-components-ts';
+const ChecklistTitleWrapper = styled.div`
```
Comment (munkhjin0223) [inline src/modules/checklists/styles.ts]: add space

File: src/modules/checklists/types.ts [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+) => Promise<any>;
+
+// checklists items
+
+export interface IChecklistItemDoc {
+  checklistId: string;
+  isChecked?: boolean;
+  content: string;
+  mentionedUserIds?: string[];
+  order?: number;
```
Comment (munkhjin0223) [inline src/modules/checklists/types.ts]: delete

File: src/modules/common/components/SortableList.tsx [hunk 16]
```diff
@@ -13,6 +13,7 @@ type Props = {
   showDragHandler?: boolean | true;
   isDragDisabled?: boolean;
   droppableId?: string;
+  isChecklistItem?: boolean;
```
Comment (munkhjin0223) [inline src/modules/common/components/SortableList.tsx]: delete

File: src/modules/common/components/SortableList.tsx [hunk 17]
```diff
@@ -54,7 +54,13 @@ class SortableList extends React.Component<Props> {
   }
 
   render() {
-    const { fields, child, isDragDisabled, droppableId } = this.props;
+    const {
+      fields,
+      child,
+      isDragDisabled,
+      droppableId,
+      isChecklistItem
```
Comment (munkhjin0223) [inline src/modules/common/components/SortableList.tsx]: delete

File: src/modules/common/components/SortableList.tsx [hunk 18]
```diff
@@ -81,6 +87,7 @@ class SortableList extends React.Component<Props> {
                       {...dragProvided.draggableProps}
                       {...dragProvided.dragHandleProps}
                       isDragging={snapshot.isDragging}
+                      isChecklistItem={isChecklistItem}
```
Comment (munkhjin0223) [inline src/modules/common/components/SortableList.tsx]: delete

File: src/modules/deals/graphql/queries.ts [hunk 19]
```diff
@@ -80,6 +80,10 @@ export const dealFields = `
   modifiedBy
   reminderMinute
   isComplete
+  checklistsState {
+    complete
+    all
+  }
```
Comment (munkhjin0223) [inline src/modules/deals/graphql/queries.ts]: delete

File: src/modules/tasks/graphql/queries.ts [hunk 20]
```diff
@@ -77,6 +77,10 @@ export const taskFields = `
   modifiedBy
   reminderMinute
   isComplete
+  checklistsState {
+    complete
+    all
+  }
```
Comment (munkhjin0223) [inline src/modules/tasks/graphql/queries.ts]: delete

File: src/modules/tickets/graphql/queries.ts [hunk 21]
```diff
@@ -80,6 +80,10 @@ export const ticketFields = `
   modifiedBy
   reminderMinute
   isComplete
+  checklistsState {
+    complete
+    all
+  }
```
Comment (munkhjin0223) [inline src/modules/tickets/graphql/queries.ts]: delete

File: src/modules/checklists/types.ts (hunk 15 above)
Comment (Daarii) [reply src/modules/checklists/types.ts]: ene 4 moriig 4vvleng ni ustga gesen vg vv? Zowhon isChecked - iig ustgaad bolhuu

**Pull request merged**